e2e-tests/tfm/cypress/e2e/journeys/case-amendments/add-bank-decision/add-bank-decision-proceed.spec.js

```diff
@@ -5,21 +5,37 @@ import facilityPage from '../../../pages/facilityPage';
 import amendmentsPage from '../../../pages/amendments/amendmentsPage';
 import caseDealPage from '../../../pages/caseDealPage';
 import MOCK_DEAL_AIN from '../../../../fixtures/deal-AIN';
-import { oneMonth, tomorrow, yearWithZeroLetter } from '../../../../../../e2e-fixtures/dateConstants';
+import { oneMonth, today, tomorrow, yearWithZeroLetter } from '../../../../../../e2e-fixtures/dateConstants';
 import { ADMIN, BANK1_MAKER1, PIM_USER_1, UNDERWRITER_MANAGER_1, UNDERWRITER_MANAGER_DECISIONS } from '../../../../../../e2e-fixtures';
 import pages from '../../../pages';
 import { CURRENCY } from '../../../../../../e2e-fixtures/constants.fixture';
 
 context('Amendments underwriting - add banks decision - proceed', () => {
-  // If the expiry & commencement date are the same day of the month then we add one to the month
-  // difference for BS (but not for EWCS)
-  // BUT... If the commencement date is end of month and the expiry date isn't then we don't add one
-  //
-  // In these tests, the mock start date is two years ago today & the mock end date is a month today.
-  // Therefore if today is EOM and a month from now is not EOM we need to NOT add one to the tenor
-  const todayIsEndOfMonth = add(new Date(), { days: 1 }).getDate() === 1;
-  const aMonthFromNowIsEndOfMonth = add(new Date(), { months: 1, days: 1 }).getDate() === 1;
-  const facilityTenor = todayIsEndOfMonth && !aMonthFromNowIsEndOfMonth ? '25 months' : '26 months';
+  /**
+   * The deal in the test is a BS deal.
+   *
+   * The tenor is calculated using the SQL here:
+   * https://github.com/UK-Export-Finance/databases-datateam/blob/163d71b0aef58652db5e6ccf29edce2ce3df30ca/MASTER_DATA/dbo/Functions/fn_DATEDIFF_FULLMONTH.sql
+   *
+   * In these tests, the mock start date is two years ago today & the mock end date is a month today.
+   *
+   * So in this case it boils down to the tenor being 25 months except for:
+   * - if expiry and commencement dates are the same date of month then you add one
+   * - if today is end of month and a month from now is also end of month you add one
+   */
+  let facilityTenor = '25 months';
+
+  if (today.date.getDate() === oneMonth.date.getDate()) {
+    facilityTenor = '26 months';
+  }
+
+  const isTodayEndOfMonth = add(today.date, { days: 1 }).getDate() === 1;
+  const isMonthFromNowEndOfMonth = add(today.date, { months: 1, days: 1 }).getDate() === 1;
+
+  if (isTodayEndOfMonth && isMonthFromNowEndOfMonth) {
+    facilityTenor = '26 months';
+  }
+
   const updatedFacilityTenor = '25 months';
```

Comment:
The test using this `updatedFacilityTenor` is used when we change the cover end date from being a month today to being tomorrow. 

I believe this test will fail once every 4 years on the 28th of February on a Leap year as on that day: 

- Tomorrow will be the end of month (29th feb)
- 2 years ago today was also the end of the month 

and so we should add one (making it 26th months)

Not too worried given how rarely this will fail.

Reply:
Resolving as not invested in this!

Reply:
Ah good point! I have changed the test to change the facility cover end date to twoDays instead of tomorrow so that this should never fail.